File: examples/python/advanced/error-handling/step-interrupted-error.py
from aws_durable_execution_sdk_python import StepInterruptedError

# This can happen if Lambda times out during step execution
@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    try:
        result = context.step(long_running_operation())
        return result
    except StepInterruptedError as e:
        # Step was interrupted, will retry on next invocation
        context.logger.warning(f"Step interrupted: {e.step_id}")
        raise

File: examples/java/advanced/error-handling/step-interrupted-error.java
// This can happen if Lambda times out during step execution
try {
    var result = ctx.step("long_running", Map.class,
        stepCtx -> longRunningOperation());
    return result;
} catch (StepInterruptedException e) {
    // Step was interrupted, will retry on next invocation
    ctx.logger().warning("Step interrupted: " + e.getStepId());
    throw e;
}
